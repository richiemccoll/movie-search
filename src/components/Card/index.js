import React from "react";
import PropTypes from "prop-types";

const getThumbsRating = (rating) => {
  if (rating > 7) {
    return `👍👍`;
  }

  if (rating >= 4 && rating <= 7) {
    return `👍`;
  }

  return `👎`;
};

export default function Card({
  image, title, overview, popularity, releaseDate,
}) {
  return (
    <article>
      <img src={image} alt="" />
      <h2>{title}</h2>
      <p>{overview}</p>
      <label className="tag release-date">Release Date - {releaseDate}</label>
      <small className="tag popularity">
        Popularity - {popularity} / 10 {getThumbsRating(popularity)}{" "}
      </small>
      <style jsx>
        {`
          article {
            max-width: 330px;
            width: 100%;
            padding-bottom: 0.5em;
            box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
          }

          h2 {
            font-size: 24px;
            font-weight: bold;
            padding: 1em;
            padding-bottom: 0;
            width: 280px;
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
          }

          img {
            width: 100%;
            object-fit: fill;
            min-height: 320px;
            max-height: 320px;
          }

          img:before {
            content: "We're sorry, the image below is broken :(";
            display: block;
            margin-bottom: 10px;
          }

          img:after {
            content: "(url: " attr(src) ")";
            display: block;
            font-size: 12px;
          }

          p {
            margin: 0 auto;
            -webkit-box-orient: vertical;
            overflow: hidden;
            text-overflow: ellipsis;
            -webkit-line-clamp: 2;
            line-height: 1.2em;
            font-size: 14px;
            height: 20px;
            display: block;
            display: -webkit-box;
            max-width: 400px;
            margin-bottom: 20px;
            padding: 1em;
          }

          .tag {
            display: block;
            padding: 1em;
            border-radius: 5px;
            font-size: 12px;
            color: white;
            width: 50%;
            margin: 1em;
            text-align: center;
          }

          .release-date {
            background-color: #ff9100;
          }

          .popularity {
            background-color: #00e5ff;
          }
        `}
      </style>
    </article>
  );
}

Card.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  popularity: PropTypes.number.isRequired,
  releaseDate: PropTypes.string.isRequired,
};
