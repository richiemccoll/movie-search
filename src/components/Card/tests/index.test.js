import React from "react";
import renderer from "react-test-renderer";

import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Card from "..";

Enzyme.configure({ adapter: new Adapter() });

describe("Card", () => {
  it("should render correctly", () => {
    const initialProps = {
      image: "url-to-image",
      title: "test",
      overview: "this is a test",
      popularity: 10,
      releaseDate: "today",
    };
    const component = renderer.create(<Card {...initialProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
