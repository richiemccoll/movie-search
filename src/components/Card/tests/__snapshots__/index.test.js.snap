// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<article
  className="jsx-127806560"
>
  <img
    alt=""
    className="jsx-127806560"
    src="url-to-image"
  />
  <h2
    className="jsx-127806560"
  >
    test
  </h2>
  <p
    className="jsx-127806560"
  >
    this is a test
  </p>
  <label
    className="jsx-127806560 tag release-date"
  >
    Release Date - 
    today
  </label>
  <small
    className="jsx-127806560 tag popularity"
  >
    Popularity - 
    10
     / 10 
    👍👍
     
  </small>
</article>
`;
